torch/testing/_internal/opinfo/definitions/linalg.py

```diff
@@ -1172,6 +1249,26 @@ def make_input():
             ),
         ),
     ),
+    OpInfo(
+        "linalg.diagonal",
+        aten_name="lingalg_diagonal",
+        aten_backward_name="diagonal_backward",
+        dtypes=all_types_and_complex_and(
+            torch.bool, torch.bfloat16, torch.float16, torch.chalf
+        ),
+        supports_out=False,
+        supports_forward_ad=True,
+        supports_fwgrad_bwgrad=True,
+        sample_inputs_func=sample_inputs_diagonal,
+        error_inputs_func=error_inputs_diagonal,
+        skips=(
+            DecorateInfo(
+                unittest.skip("Skipped"),
+                "TestVmapOperatorsOpInfo",
+                "test_vmap_linalg_failure_1D_input",
+            ),
```

Comment:
This seems like a bug on the functorch side:
```python
In [1]: import torch

In [2]: from torch import vmap

In [3]: a = torch.randn(3, 3)

In [4]: torch.linalg.diagonal(a)
Out[4]: tensor([ 0.4697, -1.6019, -0.6110])

In [5]: vmap(torch.linalg.diagonal(a))
Out[5]: <function torch.vmap.<locals>.wrapped>

In [6]: vmap(torch.linalg.diagonal)(a)
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
Input In [6], in <cell line: 1>()
----> 1 vmap(torch.linalg.diagonal)(a)

File ~/Documents/ldev/pytorch/torch/_functorch/vmap.py:434, in vmap.<locals>.wrapped(*args, **kwargs)
    430     return _chunked_vmap(func, flat_in_dims, chunks_flat_args,
    431                          args_spec, out_dims, randomness, **kwargs)
    433 # If chunk_size is not specified.
--> 434 return _flat_vmap(
    435     func, batch_size, flat_in_dims, flat_args, args_spec, out_dims, randomness, **kwargs
    436 )

File ~/Documents/ldev/pytorch/torch/_functorch/vmap.py:39, in doesnt_support_saved_tensors_hooks.<locals>.fn(*args, **kwargs)
     36 @functools.wraps(f)
     37 def fn(*args, **kwargs):
     38     with torch.autograd.graph.disable_saved_tensors_hooks(message):
---> 39         return f(*args, **kwargs)

File ~/Documents/ldev/pytorch/torch/_functorch/vmap.py:619, in _flat_vmap(func, batch_size, flat_in_dims, flat_args, args_spec, out_dims, randomness, **kwargs)
    617 try:
    618     batched_inputs = _create_batched_inputs(flat_in_dims, flat_args, vmap_level, args_spec)
--> 619     batched_outputs = func(*batched_inputs, **kwargs)
    620     return _unwrap_batched(batched_outputs, out_dims, vmap_level, batch_size, func)
    621 finally:

IndexError: Dimension out of range (expected to be in range of [-1, 0], but got -2)
```

Reply:
If it's an error (which it looks like it is) xfail it. Also, the list of xfails for functorch ops is located in the test itself. In this case, it's in https://github.com/pytorch/pytorch/blob/e3892fd16be708ade8e6bbb31e9c0b3771d24fbb/test/functorch/test_vmap.py#L4251

Reply:
Sure, updated!